### PR TITLE
add evidence models and for outstanding verifications ui

### PR DESCRIPTION
**Pull request by harshaellanki**
IVL-180558616

### CODE REVIEW

**Comment by markgoho** [comment]
@harshaellanki this is on the verge of being closed, please get a review and merge or we'll re-open when it becomes relevant again

File: components/financial_assistance/app/controllers/financial_assistance/evidences_controller.rb [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    before_action :fetch_applicant
+    before_action :updateable?, :find_type
+
+    def update_evidence
+      update_reason = params[:verification_reason]
+      admin_action = params[:admin_action]
+      reasons_list = FinancialAssistance::Evidence::VERIFY_REASONS + FinancialAssistance::Evidence::REJECT_REASONS
+      if reasons_list.include?(update_reason)
+        verification_result = admin_verification_action(admin_action, @evidence, update_reason)
+        message = (verification_result.is_a? String) ? verification_result : "Verification successfully approved."
```
Comment (markgoho) [inline components/financial_assistance/app/controllers/financial_assistance/evidences_controller.rb:18]: Does this string appear to the end user?

File: components/financial_assistance/app/controllers/financial_assistance/evidences_controller.rb [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    end
+
+    def extend_due_date
+      @family_member = FamilyMember.find(@evidence.evidenceable.family_member_id)
+      enrollment = @family_member.family.enrollments.enrolled.first
+      if enrollment.present? && @evidence.type_unverified?
+        new_date = @evidence.verif_due_date + 30.days
+        updated = @evidence.update(:due_on => new_date)
+        if updated
+          flash[:success] = "#{@evidence.title} verification due date was extended for 30 days."
```
Comment (markgoho) [inline components/financial_assistance/app/controllers/financial_assistance/evidences_controller.rb:56]: These strings need to be in translations

File: .bash_history [hunk 3]
```diff
@@ -1 +1,10 @@
 
+history | grep dcr
```
Comment (markgoho) [inline .bash_history:2]: should this be in version control?

File: Gemfile.lock [hunk 4]
```diff
@@ -7,8 +7,8 @@ GIT
 
 GIT
   remote: https://github.com/ideacrew/aca_entities.git
-  revision: b9eaa23bd4bcb7fa7a90c9845dcfca10ff93ad4a
-  branch: trunk
+  revision: 4f184b71ddc41b5064a519a154000cab5d123a51
+  branch: ov_ui_feature_180586921
```
Comment (markgoho) [inline Gemfile.lock:11]: needs to point to `trunk`

File: components/financial_assistance/app/views/financial_assistance/applications/verifications/_admin_verification_actions.html.erb [hunk 5]
```diff
@@ -1,15 +1,17 @@
...
     </div>
-    <%= form_tag financial_assistance.application_applicant_verification_documents_update_evidence_path(application, applicant), method: :put do %>
+    <%= form_tag financial_assistance.application_applicant_evidences_update_evidence_path(application, applicant), method: :put do %>
         <%= hidden_field_tag 'applicant_id', applicant.id  %>
-        <%= hidden_field_tag 'evidence', evidence.id %>
+        <%= hidden_field_tag 'evidence_kind', evidence_kind  %>
         <%= hidden_field_tag 'admin_action', 'verify' %>
         <div class="form-group col-md-3">
-          <%= select_tag :verification_reason, options_for_select(FinancialAssistance::Document::VERIFICATION_REASONS), {prompt: ' Select Reason', required: true}%>
+          <%= select_tag :verification_reason, options_for_select(::Eligibilities::Evidence::VERIFY_REASONS), {prompt: ' Select Reason', required: true}%>
```
Comment (markgoho) [inline components/financial_assistance/app/views/financial_assistance/applications/verifications/_admin_verification_actions.html.erb:14]: you're referencing these without a translation string

File: components/financial_assistance/app/views/financial_assistance/applications/verifications/_admin_verification_actions.html.erb [hunk 6]
```diff
@@ -25,12 +27,12 @@
...
     </div>
-    <%= form_tag financial_assistance.application_applicant_verification_documents_update_evidence_path(application, applicant), method: :put do %>
+    <%= form_tag financial_assistance.application_applicant_evidences_update_evidence_path(application, applicant), method: :put do %>
         <%= hidden_field_tag 'applicant_id', applicant.id  %>
-        <%= hidden_field_tag 'evidence', evidence.id %>
+        <%= hidden_field_tag 'evidence_kind', evidence_kind  %>
         <%= hidden_field_tag 'admin_action', 'return_for_deficiency' %>
         <div class="form-group col-md-3">
-          <%= select_tag :verification_reason, options_for_select(FinancialAssistance::Document::REJECT_REASONS), {prompt: ' Select Reason', required: true}%>
+          <%= select_tag :verification_reason, options_for_select(::Eligibilities::Evidence::REJECT_REASONS), {prompt: ' Select Reason', required: true}%>
```
Comment (markgoho) [inline components/financial_assistance/app/views/financial_assistance/applications/verifications/_admin_verification_actions.html.erb:35]: no translations

File: components/financial_assistance/app/views/financial_assistance/applications/verifications/_admin_verification_actions.html.erb [hunk 7]
```diff
@@ -40,3 +42,111 @@
...
+    <%end%>
+  </div>
+</div>
+<!-- ==== Extend request ==== -->
+
+<!-- ***** View History ***** -->
+<div class="row">
+  <div class="view-history col-md-12" id=<%="#{applicant.id}-#{evidence_type.downcase.split.join('-')}-history"%>>
+    <div>
+      <h5>Verification History</h5>
```
Comment (markgoho) [inline components/financial_assistance/app/views/financial_assistance/applications/verifications/_admin_verification_actions.html.erb:89]: Needs to be translated

File: components/financial_assistance/app/views/financial_assistance/applications/verifications/_admin_verification_actions.html.erb [hunk 8]
```diff
@@ -40,3 +42,111 @@
...
+      <table class="table table-bordered table-hover table-responsive">
+        <thead>
+        <tr>
+          <th scope="col">#</th>
```
Comment (markgoho) [inline components/financial_assistance/app/views/financial_assistance/applications/verifications/_admin_verification_actions.html.erb:93]: Column headings need to be translated

File: components/financial_assistance/app/views/financial_assistance/applications/verifications/_admin_verification_actions.html.erb [hunk 9]
```diff
@@ -40,3 +42,111 @@
...
+                    <h4 class="modal-title"><%= "#{evidence.title.capitalize} payload" %></h4>
+                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                      <span aria-hidden="true">&times;</span>
+                    </button>
+                  </div>
+                  <div class="modal-body">
+                    <%= JSON.parse(record.raw_payload) if record.raw_payload.present? %>
+                  </div>
+                  <div class="modal-footer">
+                    <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
```
Comment (markgoho) [inline components/financial_assistance/app/views/financial_assistance/applications/verifications/_admin_verification_actions.html.erb:137]: Missing translation

File: components/financial_assistance/config/brakeman.ignore [hunk 10]
```diff
@@ -1,5 +1,45 @@
 {
   "ignored_warnings": [
+    {
```
Comment (markgoho) [inline components/financial_assistance/config/brakeman.ignore:3]: What is the purpose of ignoring these brakeman rules?

File: components/financial_assistance/spec/dummy/app/models/eligibilities/evidence.rb [hunk 11]
```diff
@@ -0,0 +1,195 @@
+# frozen_string_literal: true
```
Comment (markgoho) [inline components/financial_assistance/spec/dummy/app/models/eligibilities/evidence.rb:1]: is this duplicate code?

File: components/financial_assistance/spec/dummy/app/models/eligibilities/family_eligibility.rb [hunk 12]
```diff
@@ -0,0 +1,200 @@
+# frozen_string_literal: true
```
Comment (markgoho) [inline components/financial_assistance/spec/dummy/app/models/eligibilities/family_eligibility.rb:1]: is this duplicate code?

File: components/financial_assistance/spec/models/financial_assistance/application_spec.rb [hunk 13]
```diff
@@ -658,6 +658,7 @@
   end
 
   describe '.create_evidences' do
+    let(:income) { FactoryBot.build(:financial_assistance_income, amount: 200, start_on: Date.new(2021,6,1), end_on: Date.new(2021, 6, 30), frequency_kind: "biweekly") }
```
Comment (markgoho) [inline components/financial_assistance/spec/models/financial_assistance/application_spec.rb:661]: why are we hard-coding 2021?

File: config/brakeman.ignore [hunk 14]
```diff
@@ -777,6 +777,26 @@
       "confidence": "Weak",
       "note": ""
     },
+    {
```
Comment (markgoho) [inline config/brakeman.ignore:780]: Why are we ignoring these brakeman errors?

**Review by markgoho — DISMISSED** [review]
Overall this PR is lacking translated english strings.

There are also handful of other questions/things to be addressed.

**Review by ipublic — APPROVED** [review]

**Pull request merged**